ui/actions/nuxeo-move-content-to-coldstorage-button.js

```diff
@@ -0,0 +1,169 @@
+import '@polymer/polymer/lib/elements/dom-if.js';
+import '@polymer/polymer/lib/elements/dom-repeat.js';
+import { FiltersBehavior } from '../nuxeo-filters-behavior.js';
+import { FormatBehavior } from '../nuxeo-format-behavior.js';
+import '../nuxeo-icons.js';
+import '../widgets/nuxeo-dialog.js';
+import '../widgets/nuxeo-tooltip.js';
+import './nuxeo-action-button-styles.js';
+
+{
+  /**
+   * A button element to move the document's content to the cold storage.
+   *
+   * Example:
+   *
+   *    <nuxeo-move-content-to-coldstorage-button document="[[document]] user="[[user]]"">
+   *    </nuxeo-move-content-to-coldstorage-button>
+   *
+   * @customElement
+   * @appliesMixin mixinBehaviors
+   * @polymerBehavior Nuxeo.FiltersBehavior
+   * @polymerBehavior Nuxeo.FormatBehavior
+   * @memberof Nuxeo
+   */
+  class MoveToColdStorage extends mixinBehaviors([FiltersBehavior, FormatBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style include="nuxeo-action-button-styles nuxeo-styles">
+        </style>
+        <nuxeo-operation id="opMove" op="Document.MoveToColdStorage" input="[[document.uid]]"></nuxeo-operation>
+
+        <dom-if if="[[_canMoveToColdStorage(document)]]">
+          <template>
+            <div class="action" on-click="_toggleDialog" hidden$="[[!_canMoveToColdStorage(document)]]">
+              <paper-icon-button icon="nuxeo:coldstorage" noink></paper-icon-button>
+              <span class="label">[[i18n('documentContent.moveToColdStorage.label')]]</span>
+              <nuxeo-tooltip>[[i18n('documentContent.moveToColdStorage.tooltip')]]</nuxeo-tooltip>
+            </div>
+          </template>
+        </dom-if>
+
+        <nuxeo-dialog id="content_to_move_dialog" with-backdrop>
+          <div>
+            <h2>[[i18n('documentContent.moveToColdStorage.ask.confirmation')]]</h2>
+          </div>
+          <p class="horizontal">
+             [[i18n('documentContent.moveToColdStorage.description')]]
+          </p>
+          </div>
+
+          <div class="buttons">
+           <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+             [[i18n('documentContent.moveToColdStorage.cancel')]]
+            </paper-button>
+           <paper-button id="confirm" name="confirm" class="primary" noink dialog-confirm on-tap="_toggle">
+              [[i18n('documentContent.moveToColdStorage.confirm')]]
+           </paper-button>
+          </div>
+        </nuxeo-dialog>
+
+         <nuxeo-dialog id="cannot_move_content_dialog" with-backdrop>
+          <p class="horizontal">
+             [[i18n('documentContent.moveToColdStorage.error.cannotPerformMove')]]
+          </p>
+          </div>
+
+          <div class="buttons">
+           <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+             [[i18n('documentContent.moveToColdStorage.ok')]]
+           </paper-button>
+          </div>
+        </nuxeo-dialog>
```

Comment:
Not sure if we need this.
In fact I believe we can assume that if we see the button is because we have permissions to use it, to use the operation. If something goes wrong, we sent the `notify` event.
For this reason I would probably drop this. And also, it is not a pattern that we use on actions.

Reply:
This probably lead us to don't factor the availability code and make everything simpler 🙂

Reply:
yes, i will remove the dialogue once and i will replace it by a:
```
this.dispatchEvent(
        new CustomEvent('notify', {
          composed: true,
          bubbles: true,
          detail: { message: [[this.i18n('documentContent.moveToColdStorage.error.cannotPerformMove')]] },
        }),
      );
```